### PR TITLE
feat(exports): decode unicode escapes in batch export pipeline

### PR DESCRIPTION
**Pull request by toyayuto**
## What does this PR do?

  Apply `decodeUnicodeEscapesOnly()` to the batch export `stringify` and `stringifyForCsv` functions so that `\uXXXX` sequences (produced by Python SDK's `json.dumps` with `ensure_ascii=True`)
   are decoded to their original characters in exported CSV/JSON/JSONL files.

  This follows the same approach already used in the Web UI (PR #9686) where `decodeUnicodeEscapesOnly()` was added to `IOTableCell` for display.

  Partially addresses #10972

  ### Changed files

  | File | Change |
  |------|--------|
  | `packages/shared/src/utils/unicode.ts` | New — `decodeUnicodeEscapesOnly()` in shared package (same implementation as `web/src/utils/unicode.ts`; Web UI can be refactored to import from
  here in the future) |
  | `packages/shared/src/server/utils/transforms/stringify.ts` | Modified — apply decode in replacer and CSV string path |
  | `worker/src/__tests__/stringify.test.ts` | New — 10 unit tests |

  ## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)

  ## Mandatory Tasks

  - [x] Make sure you have self-reviewed the code.

  ## Checklist

  - [x] My code follows the style guidelines of this project (`pnpm run format`)
  - [x] I have checked if my changes generate no new warnings (`pnpm run lint`)
  - [x] I have added tests that prove my fix is effective
  - [x] New and existing unit tests pass locally with my changes
  - [x] I have verified the fix end-to-end (exported CSV with Japanese characters before/after)

### CODE REVIEW

**Comment by nimarb** [comment]
@claude review

File: packages/shared/src/utils/unicode.ts
Comment (nimarb) [inline packages/shared/src/utils/unicode.ts:1]: rather than having this file twice in the repo, we should move the previous file to this location and import it from `shared` also in the frontend.

File: worker/src/__tests__/stringify.test.ts
Comment (nimarb) [inline worker/src/__tests__/stringify.test.ts:1]: we have tests for the other `unicode` file, no need to maintain twice if we move the file to `shared`

**Comment by nimarb** [comment]
@claude review

File: packages/shared/src/server/utils/transforms/stringify.ts [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
   const indent = key === "comments" ? 2 : undefined;
 
   return JSON.stringify(
     data,
-    (k, value) =>
-      typeof value === "bigint" ? Number.parseInt(value.toString()) : value,
+    (k, value) => {
+      if (typeof value === "bigint") return Number.parseInt(value.toString());
+      if (typeof value === "string")
+        return decodeUnicodeEscapesOnly(value, true);
```
[the remark this answers is not in the document]
Comment (nimarb) [reply packages/shared/src/server/utils/transforms/stringify.ts:13]: @toyayuto could you test this / add a test for this / check if we have a proper test for this in place? thank you !

[the remark this answers is not in the document]
Comment (toyayuto) [reply packages/shared/src/server/utils/transforms/stringify.ts:13]: @nimarb Tested and fixed. Here's what I found:

  The greedy mode had a lone-surrogate bug: `decodeUnicodeEscapesOnly("\\uD83D", true)` was emitting a WTF-16 lone surrogate (U+D83D) instead of preserving the literal `\uD83D`. This would
  corrupt to U+FFFD on UTF-8 write.

  Root cause: `tryDecodeSurrogatePair` doesn't handle double-escaped backslashes in greedy mode, so it returns null for valid pairs, and there was no surrogate-range guard on the fallback
  path.

  Fix:
  - Replaced greedy-mode surrogate handling with backslash-aware decoding
  - Added lone-surrogate guard to preserve literal `\uXXXX` (same as non-greedy mode)

  Tests added:
  - Lone high surrogate → preserves `\uD83D`
  - Lone low surrogate → preserves `\uDE00`
  - High surrogate + non-surrogate → preserves literal
  - Backslashes between surrogates edge case
